### PR TITLE
Move hooks to internal module

### PR DESCRIPTION
**Pull request by hvergara**
*Description of changes:*
Move special hooks into a separate module: `internal`

`import { useAuth } from '@aws-amplify/ui-react` => `import { useAuth } from '@aws-amplify/ui-react/internal`

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by wlee221 — APPROVED** [review]
LGTM, I'm good to merge this once docs test passes

File: packages/react/src/hooks/useAuth.ts [hunk 1]
```diff
@@ -15,28 +15,6 @@ export interface UseAuthResult {
   fetch?: () => void;
 }
 
-/**
- * React hook for Amplify Auth.
```
Comment (reesscot) [inline packages/react/src/hooks/useAuth.ts:19]: Any reason for removing the useage comment? Seems useful.

**Review by reesscot — APPROVED** [review]
💯

File: packages/react/src/hooks/useAuth.ts (hunk 1 above)
Comment (hvergara) [reply packages/react/src/hooks/useAuth.ts:19]: API is not public

**Pull request merged**